wolfssl/wolfcrypt/asn.h

```diff
@@ -1947,13 +1953,22 @@ struct Signer {
 #endif /* IGNORE_NAME_CONSTRAINTS */
     byte    subjectNameHash[SIGNER_DIGEST_SIZE];
                                      /* sha hash of names in certificate */
+    #ifdef HAVE_OCSP
+        byte    issuerNameHash[SIGNER_DIGEST_SIZE];
+                                     /* sha hash of issuer names in certificate.
+                                      * Used in OCSP to check for authorized
+                                      * responders. */
+    #endif
     #ifndef NO_SKID
         byte    subjectKeyIdHash[SIGNER_DIGEST_SIZE];
-                                     /* sha hash of names in certificate */
+                                     /* sha hash of key in certificate */
     #endif
     #ifdef HAVE_OCSP
         byte subjectKeyHash[KEYID_SIZE];
     #endif
+#ifdef WOLFSSL_AKID_NAME
+    byte serialHash[SIGNER_DIGEST_SIZE]; /* serial number hash */
```

Comment:
Why serialHash when WOLFSSL_AKID_NAME?

Reply:
We only use the `serialHash` when we are matching on the `authorityCertIssuer` and the `authorityCertSerialNumber` of the AuthorityKeyIdentifier extension. That is why it is guarded by `WOLFSSL_AKID_NAME`.